### PR TITLE
Feature/range mappings

### PR DESCRIPTION
**Pull request by Mpdreamz**
This adds support for the `*_range` data types introduced in Elasticsearch 5.2

### CODE REVIEW

File: src/Nest/Mapping/Types/FieldType.cs [hunk 1]
```diff
@@ -69,5 +61,35 @@ public enum FieldType
...
+		[EnumMember(Value = "integer_range")]
+		IntegerRange,
+		[EnumMember(Value = "float_range")]
+		FloatRange,
+		[EnumMember(Value = "long_range")]
+		LongRange,
+		[EnumMember(Value = "double_range")]
+		DoubleRange,
+		[EnumMember(Value = "date_range")]
+		DateRange,
```
Comment (russcam) [inline src/Nest/Mapping/Types/FieldType.cs]: This also needs to handle custom field types e.g. [Rosette's `name` data type](https://www.rosette.com/elastic/)

Comment (Mpdreamz) [inline src/Nest/Mapping/Types/FieldType.cs]: The converter did not understand custom IProperty when they were string either. We can ship with an abstract `PluginPropertyBase` that opens it back up.

Comment (russcam) [inline src/Nest/Mapping/Types/FieldType.cs]: Sounds like a plan 👍

Comment (Mpdreamz) [inline src/Nest/Mapping/Types/FieldType.cs]: Amended PR with `PluginPropertyBase` and test

Comment (Mpdreamz) [inline src/Nest/Mapping/Types/FieldType.cs]: @forloop thumbs up counts as PR approval? :)

Comment (russcam) [inline src/Nest/Mapping/Types/FieldType.cs]: @Mpdreamz yes 😄👍

**Comment by Mpdreamz** [comment]
ported to `master`

**Pull request merged**